include/private/tdycoreimpl.h

```diff
@@ -61,26 +61,27 @@ struct _p_TDy {
   PetscReal  Pref;       /* reference pressure */
   PetscReal  Tref;       /* reference temperature */
 
-
   /* material parameters */
-  PetscReal *Sr;                 /* residual saturation (min) [1] */
-  PetscReal *K,
-            *K0;                 /* permeability tensor (cell,intrinsic) for each cell [m2] */
-  PetscReal *Kr, *dKr_dS;        /* relative permeability for each cell [1] */
-  PetscReal *porosity;           /* porosity for each cell [1] */
-  PetscReal *S,
-            *dS_dP,              /* saturation and derivative wrt pressure for each cell [1] */
-            *d2S_dP2,            /* second derivative of saturation wrt pressure for each cell [1] */
-            *dS_dT;
-  PetscReal *Kappa,
-            *Kappa0;             /* thermal conductivity tensor (cell,intrinsic) for each cell [W/(K-m)] */
-  PetscReal *Cr;                 /* specific heat capacity for rock [J/(kg-K)] */
-  PetscReal *rhor;               /* rock density [kg/m3] */
-
+  PetscReal *matprop_K,
+            *matprop_K0;         /* permeability tensor (cell,intrinsic) for each cell [m2] */
+  PetscReal *matprop_porosity;   /* porosity for each cell [1] */
+  PetscReal *matprop_Kappa,
+            *matprop_Kappa0;     /* thermal conductivity tensor (cell,intrinsic) for each cell [W/(K-m)] */
+  PetscReal *matprop_Cr;         /* specific heat capacity for rock [J/(kg-K)] */
+  PetscReal *matprop_rhor;       /* rock density [kg/m3] */
```

Comment:
Create a struct for this, say `struct _p_TDyMatProps`. This will avoid adding a prefix `matprop_`.